src/main/java/edu/umich/its/cpm/MigrationTaskService.java

```diff
@@ -1454,8 +1455,49 @@ private String generateResourcesZipFileName(String site_name) {
 			site_name = Utils.sanitizeName(site_name);
 			String zipFileName = site_name + "_content.zip";
 			log.debug("resourceZipFileName: [{}]",zipFileName);
+			
 			return zipFileName;
 		}
+		/**
+		 * construct the Content-Disposition String based on fileName
+		 * ready to be included in the response header
+		 * @param fileName
+		 * @return
+		 */
+		private String contentDispositionString(String fileName)
+		{
+			// if the fileName contains non-ASCII characters, we need to encode it
+			if (!isASCII(fileName))
+			{
+				try
+				{
+					fileName = URLEncoder.encode(fileName, "UTF-8");
+				}
+				catch (UnsupportedEncodingException e)
+				{
+					log.warn("Cannot URLEncode ZipFileName " + fileName);
+				}
+			}
+			
+			// construct the string value according to https://tools.ietf.org/html/rfc6266
+			return "attachment;filename=\"" + fileName + "\"; filename*=utf-8''" + fileName;
+		}
+
+		/**
+		 * check whether sValue contains only ASCII characters
```

Comment:
Consider the apache commons-lang-3 utility.  I'm sure it is well tested.

E.g. System.out.println(StringUtils.isAsciiPrintable("!@£$%^&!@£$%^"));